promises.cf

```diff
@@ -131,6 +126,59 @@ bundle common inventory
 
       #
 
+bundle common cfe_internal_inputs
+{
+  vars:
+    any::
+
+      "input[cfe_internal_management]"
+        string => "cfe_internal/CFE_cfengine.cf",
+        comment => "This policy activates internal managment policies
+                    for both core and enterprise";
+
+      "input[core_main]"
+        string => "cfe_internal/core/main.cf",
+        comment => "This policy activates other core policies";
+
+      "input[core_limit_robot_agents]"
+        string => "cfe_internal/core/limit_robot_agents.cf",
+        comment => "The policy here ensures that we don't have too many
+                    cf-monitord or cf-execd processes";
+
+      "input[core_log_rotation]"
+        string => "cfe_internal/core/log_rotation.cf",
+        comment => "This policy ensures that various cfengine log files
+                    do not grow without bound and fill up the disk";
+
+    !cfengine_3_5::
+      "input[core_host_info_report]"
+        string => "cfe_internal/core/host_info_report.cf",
+        comment => "This policy produces a text based host info report
+                    and serves as a functional example of using mustache templates";
+
+    enterprise_edition::
+
+      "input[enterprise_main]"
+        string => "cfe_internal/enterprise/main.cf",
+        comment => "This policy activates other enterprise specific policies";
+
+      "input[enterprise_main]"
+        string => "cfe_internal/enterprise/main.cf",
+        comment => "This policy activates other enterprise specific policies";
+
+      "input[enterprise_hub_specific]"
+        string => "cfe_internal/enterprise/CFE_hub_specific.cf",
+        comment => "Policy relating to CFEngine Enterprise Hub, for example
+                    software updates, webserver configuration, and alerts";
+      
+      "input[enterprise_knowledge]"
+        string => "cfe_internal/enterprise/CFE_knowledge.cf",
+        comment => "Settings mostly releated to CFEngine Enteprise Mission Portal";
+
+    any::
+      "inputs" slist => getvalues("input");
```

Comment:
I'm glad you like it, I thought it was clever too, and IMHO a good use of classic arrays.